### PR TITLE
feat: support timeout and failOpen/failClose knobs for Global RateLimiting

### PR DESCRIPTION
**Pull request by tmsnan**
Support timeout and failOpen/failClose knobs for Global RateLimiting
releated #1655

### CODE REVIEW

File: api/config/v1alpha1/envoygateway_types.go [hunk 1]
```diff
@@ -273,6 +273,17 @@ type RateLimit struct {
...
+	// Timeout specifies the timeout period for the proxy to access the ratelimit server
+	// If not set, timeout is 20ms.
+	Timeout string `json:"timeout"`
+
+	// FailOpen is a switch used to control the flow of traffic
+	// when the response from the ratelimit server cannot be obtained.
+	// If FailOpen is true, let the traffic pass,
+	// otherwise, don't let the traffic pass and return 500.
+	// If not set, FailOpen is False.
+	FailOpen bool `json:"failOpen"`
```
Comment (arkodg) [inline api/config/v1alpha1/envoygateway_types.go:286]: imo, should be changed to `FailClosed` which is `false` by default, so by default its `FailOpen`

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by Xunzhuo — APPROVED** [review]
/lgtm, thanks!

**Comment by zirain** [comment]
cc @arkodg, should this landing in v0.5?

**Comment by arkodg** [comment]
@zirain since this is an feature/API change, and rc has been released for v0.5, this cannot make it in.
Lets merge this PR once v0.5 release, since this is an API change and affects `docs` dir which is yet to be created for v0.5

**Pull request merged**